test/openssl/test_ssl.rb

```diff
@@ -564,6 +564,19 @@ def test_forbid_tls_v1_for_client
     }
   end
 
+  def test_allow_tls_v1_for_client
+    # server does not support SSLv2 / SSLv3
+    ctx_proc = Proc.new { |ctx| ctx.options = OpenSSL::SSL::OP_ALL | OpenSSL::SSL::OP_NO_SSLv3 | OpenSSL::SSL::OP_NO_SSLv2 }
+    start_server_version(:TLSv1, ctx_proc) { |server, port|
+      ctx = OpenSSL::SSL::SSLContext.new
+      # It appears that explicitly calling 'ssl_version=' directly
+      # is required rather than allowing `set_params` to call it via `__send__`
+      # ctx.ssl_version = :TLSv1
+      ctx.set_params(ssl_version: :TLSv1) # soils the ssl_version
```

Comment:
I changed this line to ```ctx.set_params({ssl_version: :TLSv1, verify_mode: OpenSSL::SSL::VERIFY_NONE})```. It passed this test. 

It shows ```set_params``` works correctly. How do you think?